osu.Game/Beatmaps/BeatmapProcessor.cs

```diff
@@ -22,34 +20,17 @@ public BeatmapProcessor(IBeatmap beatmap)
 
         public virtual void PreProcess()
         {
-            IHasComboInformation lastObj = null;
-
-            bool isFirst = true;
+            IHasComboInformation? lastObj = null;
 
             foreach (var obj in Beatmap.HitObjects.OfType<IHasComboInformation>())
             {
-                if (isFirst)
+                if (lastObj == null)
                 {
-                    obj.NewCombo = true;
-
                     // first hitobject should always be marked as a new combo for sanity.
-                    isFirst = false;
-                }
-
-                obj.ComboIndex = lastObj?.ComboIndex ?? 0;
-                obj.ComboIndexWithOffsets = lastObj?.ComboIndexWithOffsets ?? 0;
-                obj.IndexInCurrentCombo = (lastObj?.IndexInCurrentCombo + 1) ?? 0;
-
-                if (obj.NewCombo)
-                {
-                    obj.IndexInCurrentCombo = 0;
-                    obj.ComboIndex++;
-                    obj.ComboIndexWithOffsets += obj.ComboOffset + 1;
-
-                    if (lastObj != null)
-                        lastObj.LastInCombo = true;
+                    obj.NewCombo = true;
```

Comment:
Is there a particular reason why this logic is staying local to here, rather than being moved to `UpdateComboInformation()` too? `NewCombo` *is* combo information, after all, and the invariant of "first object always is `NewCombo`" is a nice one to rely on.

Reply:
It was a bit of a forced hand. By putting it in `UpdateComboInformation` it would get set to `true` on a placement blueprint, which causes it to get stuck in the true state even when it should potentially not be.

I think it makes sense to not be in there because in the 99.99% case, `NewCombo` is externally managed. The "set true for first object" really should only happen at parsing time.